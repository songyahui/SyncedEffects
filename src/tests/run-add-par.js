"use hopscript"

var hh = require("hiphop");

var m1 = <hh.module S U W Z>
  <hh.parallel>
    <hh.emit W ifApply=${function() {return this.S.now}}/>
    <hh.emit Z ifApply=${function() {return this.U.now}}/>
  </hh.parallel>
</hh.module>;

var run2 = <hh.module S U A B>
  <hh.parallel id="par">
    <hh.run module=${m1} S U W=A Z=B/>
    <hh.halt/>
  </hh.parallel>
</hh.module>;

var m = new hh.ReactiveMachine(run2, "run2");
m.debug_emitted_func = console.log

//console.log(m.pretty_print());
console.log("m.inputAndReact(S)");
m.inputAndReact("S")

//m.react();
m.getElementById("par").appendChild(<hh.run module=${m1} S U Z=A/>);

console.log("==================== ADD RUN PARALLEL ==================");

//console.log(m.pretty_print());
console.log("m.inputAndReact(U)");
m.inputAndReact("U")
