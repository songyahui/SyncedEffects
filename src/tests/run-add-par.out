m.inputAndReact(S)
[ 'S', 'A' ]
==================== ADD RUN PARALLEL ==================
m.inputAndReact(U)
[ 'U', 'A' ]
